torch/nn/parallel/data_parallel.py

```diff
@@ -145,27 +145,28 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
             self.module.to(self.src_device_obj)
 
     def forward(self, *inputs, **kwargs):
-        if not self.device_ids:
-            return self.module(*inputs, **kwargs)
-
-        for t in chain(self.module.parameters(), self.module.buffers()):
-            if t.device != self.src_device_obj:
-                raise RuntimeError("module must have its parameters and buffers "
-                                   "on device {} (device_ids[0]) but found one of "
-                                   "them on device: {}".format(self.src_device_obj, t.device))
-
-        inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
-        # for forward function without any inputs, empty list and dict will be created
-        # so the module can be executed on one device which is the first one in device_ids
-        if not inputs and not kwargs:
-            inputs = ((),)
-            kwargs = ({},)
-
-        if len(self.device_ids) == 1:
-            return self.module(*inputs[0], **kwargs[0])
-        replicas = self.replicate(self.module, self.device_ids[:len(inputs)])
-        outputs = self.parallel_apply(replicas, inputs, kwargs)
-        return self.gather(outputs, self.output_device)
+        with torch.autograd.profiler.record_function("DataParallel.forward"):
```

Comment:
Does this record function regardless of whether profiler is otherwise active? Is it desired effect?

Reply:
the record_function is only active when the profiler is enabled and it will be ignored in case of profiler is inactive.

Reply:
Please see https://github.com/pytorch/pytorch/blob/159fdde9ae4d7485a1ed9c9658ecf570590df669/aten/src/ATen/record_function.cpp#L443-L445 for more information.